### PR TITLE
Chatbox splited & Remove Vuex

### PR DESCRIPTION
**Pull request by y-nk**

### CODE REVIEW

File: src/chatbox/sdkInstance.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import EkoClient from "eko-sdk";
+
+export const ClientInstance = (() => {
+  // Private Interface
+  var _client;
+  var _apikey;
```
Comment (y-nk) [inline src/chatbox/sdkInstance.js:6]:
```suggestion
  let _client;
  let _apikey;
```

File: src/chatbox/sdkInstance.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    if ( typeof _client != "undefined") {
+      return Object.keys(_client).length > 0;
+    } 
+    return false
+  }
+
+  // Public Interface
+  return { 
+    init(key) {
+      if ( !isClientExist() && (_apikey != key) ) {
```
Comment (y-nk) [inline src/chatbox/sdkInstance.js:17]:
```suggestion
      if (!isClientExist() && (_apikey !== key) ) {
```

File: src/views/Home.vue [hunk 3]
```diff
@@ -25,13 +24,12 @@ export default {
...
+    user: "",
     channel: CHANNEL_NAME,
+    api_key: "",
   }),
-  methods: {
-    ...mapMutations(["setChannel"]),
-  },
   beforeMount() {
-    this.setChannel(this.channel);
+    this.user = this.$route.params.userId;
```
Comment (y-nk) [inline src/views/Home.vue:32]:
```suggestion
```

File: src/views/Home.vue [hunk 4]
```diff
@@ -25,13 +24,12 @@ export default {
     VideoPlayer,
   },
   data: () => ({
+    user: "",
```
Comment (y-nk) [inline src/views/Home.vue:27]:
```suggestion
```

File: src/views/Home.vue [hunk 5]
```diff
@@ -25,13 +24,12 @@ export default {
     VideoPlayer,
   },
   data: () => ({
```
Comment (y-nk) [inline src/views/Home.vue:26]:
```suggestion
  computed: {
    user: ({ $route }) => $route.params.userId
  },

  data: () => ({
```

File: src/router/routes.js [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
-    name: "Login",
+    name: "login",
     component: () =>
       import(/* webpackChunkName: "login" */ "@/views/Login.vue"),
   },
+  {
+    path: "/home/:userId",
+    name: "home",
+    component: () => import(/* webpackChunkName: "home" */ "@/views/Home.vue"),
+  },
```
Comment (y-nk) [inline src/router/routes.js:12]:
```suggestion
  {
    path: "/",
    redirect: () => {
      if (/* test to know if ClientInstance is connected */)
        return { name: `/home/${ClientInstance.currentUserId}`}
        
      return { name: "login" }
  },
  {
    path: "/login",
    name: "login",
    component: () =>
      import(/* webpackChunkName: "login" */ "@/views/Login.vue"),
  },
  {
    path: "/home/:userId",
    name: "home",
    component: () => import(/* webpackChunkName: "home" */ "@/views/Home.vue"),
  },
```

File: src/chatbox/ChatBox.vue [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        });
+        this.isJoined = true;
+      } else {
+        this.isJoined = false;
+      }
+    },
+  },
+  beforeMount() {
+    this.current_key = this.api_key;
+  },
```
Comment (y-nk) [inline src/chatbox/ChatBox.vue:56]:
```suggestion
```

File: src/chatbox/ChatBox.vue [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    current_key: "",
+    isJoined: false,
+  }),
+  watch: {
+    current_key: function (val) {
+      ClientInstance.init(val);
+      ClientInstance.registerUserSession(this.userId);
+      this.joinUserToChannel(this.channelId, EkoChannelType.Standard);
+    },
+  },
```
Comment (y-nk) [inline src/chatbox/ChatBox.vue:37]:
```suggestion
```

File: src/chatbox/ChatBox.vue [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+export default {
+  name: "ChatBox",
+  components: {
+    ChatBoxHeader,
+    ChatBoxInput,
+    ChatBoxMessageList,
+  },
+  props: ["api_key", "userId", "channelId"],
+  data: () => ({
+    current_key: "",
```
Comment (y-nk) [inline src/chatbox/ChatBox.vue:28]:
```suggestion
```

File: src/chatbox/ChatBox.vue [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+          console.log("Channel data update: ", data);
+        });
+        this.isJoined = true;
+      } else {
+        this.isJoined = false;
+      }
+    },
+  },
+  beforeMount() {
+    this.current_key = this.api_key;
```
Comment (y-nk) [inline src/chatbox/ChatBox.vue:55]:
```suggestion
      ClientInstance.init(this.api_key);
      ClientInstance.registerUserSession(this.userId);
```

File: src/chatbox/ChatBox.vue (hunk 7 above)
Comment (y-nk) [inline src/chatbox/ChatBox.vue:56]:
```suggestion
  },
  mounted() {
    this.joinUserToChannel(this.channelId, EkoChannelType.Standard);
  },
```

File: src/chatbox/ChatBox.vue [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+      if (Object.keys(liveChannel).length > 0) {
+        liveChannel.once("dataUpdated", (data) => {
+          console.log("Channel data update: ", data);
+        });
+        this.isJoined = true;
+      } else {
+        this.isJoined = false;
+      }
+    },
+  },
```
Comment (y-nk) [inline src/chatbox/ChatBox.vue:53]:
```suggestion
```

File: src/chatbox/ChatBox.vue (hunk 8 above)
Comment (y-nk) [inline src/chatbox/ChatBox.vue:37]:
```suggestion
```

File: src/chatbox/ChatBox.vue [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+        type,
+      });
+      if (Object.keys(liveChannel).length > 0) {
+        liveChannel.once("dataUpdated", (data) => {
+          console.log("Channel data update: ", data);
+        });
+        this.isJoined = true;
+      } else {
+        this.isJoined = false;
+      }
```
Comment (y-nk) [inline src/chatbox/ChatBox.vue:51]:
```suggestion
      this.isJoined = false
      
      const liveChannel = channelRepo.joinChannel({ channelId, type });
      liveChannel.once("dataUpdated", (data) => this.isJoined = true)
```

Comment (supakorn-amity) [reply src/chatbox/ChatBox.vue:51]: From the above comments that remove watch: { as well as methods: { out. So where to put the function in then in 

`  beforeMount() {
    ClientInstance.init(this.api_key);
    ClientInstance.registerUserSession(this.userId);
     this.isJoined = false
     const liveChannel = channelRepo.joinChannel({ channelId, type });
     liveChannel.once("dataUpdated", (data) => this.isJoined = true)
  },`

Comment (y-nk) [reply src/chatbox/ChatBox.vue:51]: oh shit. let me correct this error in my review.

Comment (supakorn-amity) [reply src/chatbox/ChatBox.vue:51]: Screen Shot 2564-05-18 at 10 58 05

**Pull request merged**